collections/ansible_collections/cloudkit/service/roles/cleanup_stale_network_resources/tasks/cleanup.yaml

```diff
@@ -0,0 +1,245 @@
+---
+- name: Get all existing ClusterOrders from specified namespace
+  kubernetes.core.k8s_info:
+    api_version: cloudkit.openshift.io/v1alpha1
+    kind: ClusterOrder
+    namespace: "{{ cleanup_stale_network_resources_namespace }}"
+  register: cluster_orders
+  failed_when: false
+
+- name: Initialize existing cluster names list
+  ansible.builtin.set_fact:
+    existing_cluster_names: []
+
+- name: Extract cluster names from ClusterOrders
+  ansible.builtin.set_fact:
+    existing_cluster_names: "{{ existing_cluster_names + ['clusterorder_' + item.metadata.name] }}"
```

Comment:
I think you should be able to avoid the initialization task above with something like `existing_cluster_names: "{{ existing_cluster_names | default([]) + ['clusterorder_' + item.metadata.name] }}"`